src/mcp/types.py

```diff
@@ -366,7 +446,142 @@ class ServerCapabilities(BaseModel):
     """Present if the server offers any tools to call."""
     completions: CompletionsCapability | None = None
     """Present if the server offers autocompletion suggestions for prompts and resources."""
+    tasks: ServerTasksCapability | None = None
+    """Present if the server supports task-augmented requests."""
+    model_config = ConfigDict(extra="allow")
+
+
+TaskStatus = Literal["working", "input_required", "completed", "failed", "cancelled"]
+
+
+class RelatedTaskMetadata(BaseModel):
+    """
+    Metadata for associating messages with a task.
+
+    Include this in the `_meta` field under the key `io.modelcontextprotocol/related-task`.
+    """
+
+    model_config = ConfigDict(extra="allow")
+    taskId: str
+
+
+class Task(BaseModel):
+    """Data associated with a task."""
+
+    model_config = ConfigDict(extra="allow")
+
+    taskId: str
+    """The task identifier."""
+
+    status: TaskStatus
+    """Current task state."""
+
+    statusMessage: str | None = None
+    """  
+    Optional human-readable message describing the current task state.
+    This can provide context for any status, including:
+    - Reasons for "cancelled" status
+    - Summaries for "completed" status
+    - Diagnostic information for "failed" status (e.g., error details, what went wrong)
+    """
+
+    createdAt: datetime  # Pydantic will enforce ISO 8601 and re-serialize as a string later
+    """ISO 8601 timestamp when the task was created."""
+
+    ttl: Annotated[int, Field(strict=True)] | None
+    """Actual retention duration from creation in milliseconds, null for unlimited."""
+
+    pollInterval: Annotated[int, Field(strict=True)] | None = None
```

Comment:
I think we're missing `lastUpdatedAt` from the latest spec: https://modelcontextprotocol.io/specification/draft/schema#task